### DISCUSSION TITLE
null values are discarded from OpenAPI examples

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [fastapi/fastapi#8401] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

Taken from https://fastapi.tiangolo.com/advanced/additional-responses/

```python
from fastapi import FastAPI

import json

app = FastAPI()

@app.get("/", responses={200: {"content": {"application/json": {"example": {"absent": None, "present": "value"}}}}})
def read_root():
    return {"absent": None, "present": "value"}

print(json.dumps(app.openapi()))
```

### Description

- Add an example to the OpenAPI spec (using `example` or `examples`)
- Set the value to `None`
- The example is absent

The code above outputs:

```
{"openapi": "3.0.2", "info": {"title": "FastAPI", "version": "0.1.0"}, "paths": {"/": {"get": {"summary": "Read Root", "operationId": "read_root__get", "responses": {"200": {"description": "Successful Response", "content": {"application/json": {"schema": {}, "example": {"present": "value"}}}}}}}}}
```

As you can see, the `absent` example is absent.

### Operating System

Linux

### FastAPI Version

0.73.0

### Python Version

3.8.9

### Additional Context

`openapi.utils.get_openapi` has this line:

    return jsonable_encoder(OpenAPI(**output), by_alias=True, exclude_none=True)  # type: ignore

`exclude_none` could cause the issue.

When setting it to True, OpenAPI generation fails with the following error:

    n is null

**Comment by WilliamDEdwards** [comment]
Over a year later, coming across my own issue... Which has been converted into a discussion?!

With Pydantic 2.x, there's a clear differentiation between required and optional. This causes my OpenAPI examples to be incorrect, because null values are omitted for required fields.

**Comment by cohml** [comment]
Bump. I'm experiencing the same and it's a headache. A required parameter with null as an acceptable value is not the same as an optional parameter.

This should be rectified, given the obvious interest on this discussion and the issue that spawned it. Or at the very least, implement a mechanism whereby developers can toggle whether or not display null parameters via the examples.

**Comment by ehusby** [comment]
Bump! I just encountered this issue, and tested that https://github.com/fastapi/fastapi/pull/5693 gives me the proper configuration. Would love to see this fixed in the next release!

**Comment by WilliamDEdwards** [comment]
Remember when I came across my own issue - that I made in 2022 - in 2024? Here we go again... 🤦
